perma_web/perma/views/user_management.py

```diff
@@ -999,32 +999,36 @@ def edit_user_in_group(request, user_id, group_name):
         # registrar users can edit their sponsored users,
         # and users who belong to any of their registrar's organizations
         sponsorships = target_user.sponsorships.filter(registrar=request.user.registrar)
-        orgs = target_user.organizations.all() & Organization.objects.filter(registrar=request.user.registrar)
+        user_org_affiliations = UserOrganizationAffiliation.objects.filter(user=target_user).all()
+        registrar_orgs = Organization.objects.filter(registrar=request.user.registrar)
+        affiliations = user_org_affiliations.filter(organization__in=registrar_orgs)
```

Comment:
Hmmmm. If I'm reading correctly, these are not quite the same. 
`orgs = target_user.organizations.all() & Organization.objects.filter(registrar=request.user.registrar)`
 is a sum: you get all the orgs that the user is a personally a member of, and then you add to that list all the orgs associated with request.user.registrar.

I think the revised code does something different:  `user_org_affiliations = UserOrganizationAffiliation.objects.filter(user=target_user).all()` helps find all the orgs that the user is a member of, and `Organization.objects.filter(registrar=request.user.registrar)` helps find all the orgs associated with `request.user.registrar`, but `affiliations = user_org_affiliations.filter(organization__in=registrar_orgs)`, instead of producing a big list of orgs, combining the two, instead makes the queryset of `user_org_affiliations` smaller, restricting it to only include affiliations associated with that one single registrar.

I am not sure if the original logic is entirely correct: I think we _do not allow_ registrar users to also be members of any organizations, which in practice, I think means `target_user.organizations.all()` should always be an empty queryset...

But, if that is the case, I think that also means that in practice, `affiliations = user_org_affiliations.filter(organization__in=registrar_orgs)` will always be an empty queryset.

Reply:
`orgs = target_user.organizations.all() & Organization.objects.filter(registrar=request.user.registrar)`

is making an intersection of all the organizations of the user and the organizations tied to the user (the one viewing request - registrar user in this case). So it is filtering out any organizations that this registrar isn't associated with which was what I tried to implement.

And it sounds correct to me that if a registrar is viewing a user, we want them to only see the affiliations that are under their supervision since we don't want them to have the ability to remove the affiliations that are tied to different registrars (through that registrar's organizations. But as always, correct me if my assumptions are not correct.

Reply:
Gah! So sorry, my brain was mxing up `target_user` and `request.user`.

The things I said make no sense. 

Carry on 😁